### PR TITLE
#2060 updated map_cache documentation in markdown syntax to reflect t…

### PR DESCRIPTION
**Pull request by jvastbinder**
…he new map_cache system

[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
The documentation for the map_cache system did not match with the new system. So I created a markdown document to reflect the changes as shown in this [guide](http://herc.ws/board/topic/15868-guide-mapcache-generation-2018/)

**Affected Branches:** 
Master

**Issues addressed:**
#2060 

### Known Issues and TODO List

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2274)

**Comment by Emistry** [comment]
missing header, license header, might as well consider help to apply some format wherever it applicable? like file extension , path wrapped in `code` tag
command and description split to different newline
remove unnecessary message like ... 
> Feel free to ask for help here. I'll try to answer questions re: mapcache if possible. And if you think anything needs correcting or added, let me know.

**Comment by jvastbinder** [comment]
Not 100% clear on what you meant by header and license header, but I added more formatting. Let me know what else needs to be changed

**Comment by Emistry** [comment]
may refer this too https://github.com/HerculesWS/Hercules/blob/master/doc/effect_list.md

# Copyright
> This file is part of Hercules.
> http://herc.ws - http://github.com/HerculesWS/Hercules
> 
> Copyright (C) 2012-2018 Hercules Dev Team
> Copyright (C) Athena Dev Teams
> 
> Hercules is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
> 
> This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
> See the GNU General Public License for more details.
>
> You should have received a copy of the GNU General Public License along with this program.  
> If not, see <http://www.gnu.org/licenses/>.

File: doc/map_cache.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Hercules Documentation
+By: [Myriad](http://herc.ws/board/topic/15868-guide-mapcache-generation-2018/)
+Current Version: 20180923
+Description: An updated manual for Hercules' map cache generator 
+
+##Mapcache Generation 2018
```
Comment (dastgirp) [inline doc/map_cache.md]: Space after hash

File: doc/map_cache.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+- `db/[pre-re or re]/map_cache.dat` has been dropped (no longer supported).
+- In its place are individual `.mcache` files for every map located in `maps/[pre-re or re]/`
+- `mapcache`executable has been removed.
+- Replaced with the `mapcache` plugin (`src/plugins/mapcache.c`).
+
+### How to generate the mapcache?
+
+1. Same as before, check `conf/map/maps.conf` and `db/map_index.txt` have all the maps you want to cache.
+2. Your maps need to exist somewhere in your repository! There are two ways for the plugin to find them:
+    a. Place all your maps, including `resnametable.txt`, inside the **data folder** of your Hercules repo. I.e. `Hercules/data/prontera.gat/gnd/gnd/rsw` (note: I forget if all three files are needed).
```
Comment (dastgirp) [inline doc/map_cache.md]: Start with
`- ` instead of `a.`

File: doc/map_cache.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    b. OR Configure your `conf/grf-files.txt` to tell it where to find your GRF(s) which contains your maps.
```
Comment (dastgirp) [inline doc/map_cache.md]: Same as above

File: doc/map_cache.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+A list of usable parameters will appear. Here are the ones you need to know for mapcache:
+```
+[Info]:   --convert-old-mapcache         Converts an old db/pre-re/map_cache.dat file to the new format. [Mapcache]
+[Info]:   --rebuild-mapcache             Rebuilds the entire mapcache folder (maps/pre-re/), using db/map_index.txt as index. [Mapcache]
+[Info]:   --map <name>                   Rebuilds an individual map's cache into maps/pre-re/ (usage: --map <map_name_without_extension>). [Mapcache]
+[Info]:   --fix-md5                      Updates the checksum for the files in maps/pre-re/, using db/map_index.txt as index (see PR #1981). [Mapcache]
+```
+
+- Rebuild all the `.mcache files` using your old `db/[pre-re or re]map_cache.dat` file.
```
Comment (dastgirp) [inline doc/map_cache.md]: Add colon instead of full stop. so it's clear that below command needs to be executed

**Comment by Helianthella** [comment]
is it really necessary to have the copyright info visible in the rendered file? couldn't we just have an html comment on top?

```html
<!--
...
-->
```

**Comment by EyesOfAHawk** [comment]
Please write this guide in your own words or give credit to me.
http://herc.ws/board/topic/15868-guide-mapcache-generation-2018/

**Comment by 4144** [comment]
please not create fix commits for not pushed to master changes.
if you want fix your own commits in this pull request, you should squash you commit and fix for it into one commit

**Comment by Zarbony** [comment]
When will this get into the Herc Doc folder?
does this is finally ready to get in the next Release?

**Review by MishimaHaruna — APPROVED** [review]

**Pull request merged**